tests/roles/glance_adoption/tasks/main.yaml

```diff
@@ -76,37 +76,82 @@
     '
   when: glance_backend == 'ceph'
 
-- name: wait for Glance to start up
+- name: deploy podified Glance with Swift backend
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
+    oc patch openstackcontrolplane openstack --type=merge --patch '
+    spec:
+      glance:
+        enabled: true
+        template:
+          databaseInstance: openstack
+          customServiceConfig: |
+                [DEFAULT]
+                enabled_backends = default_backend:swift
+                [glance_store]
+                default_backend = default_backend
+                [default_backend]
+                swift_store_create_container_on_put = True
+                swift_store_auth_version = 3
+                swift_store_auth_address = {{ .KeystoneInternalURL }}
+                swift_store_endpoint_type = internalURL
+                swift_store_user = service:glance
+                swift_store_key = {{ .ServicePassword }}
+          storageClass: "local-storage"
+          storageRequest: 10G
+          glanceAPIs:
+            default:
+              replicas: 0
+              override:
+                service:
+                  internal:
+                    metadata:
+                      annotations:
+                        metallb.universe.tf/address-pool: internalapi
+                        metallb.universe.tf/allow-shared-ip: internalapi
+                        metallb.universe.tf/loadBalancerIPs: 172.17.0.80
+                  spec:
+                    type: LoadBalancer
+              networkAttachments:
+              - storage
+    '
+  when: glance_backend == 'swift'
 
-    STATUS=$(oc get pod --selector=service=glance -o jsonpath='{.items[*].status.phase}{"\n"}');
-    code=1
-    IFS=" " read -r -a STATUS <<< "$STATUS"
-    for i in "${STATUS[@]}"; do
-        if echo "$i" | grep -v Running; then
-            # if at least one instance is not Running, return
-            exit $code
-        else
-            code=0
-        fi
-    done
-    exit $code
-  register: glance_running_result
-  until: glance_running_result is success
-  retries: 60
-  delay: 2
+- name: Check the resulting Glance
+  when: glance_backend != 'swift'
```

Comment:
@jistr when patch https://github.com/openstack-k8s-operators/data-plane-adoption/pull/291 lands, we can improve this part and have an additional task in `swift` to `patch` the glance replicas.

Reply:
the above is just an example, I suspect we have to both retrieve the right `osctlplane` name and pass the right path to the `check_glance.yaml` tasks.